### PR TITLE
Added support for yum compatible distro (Mainly centos/redhat); Tested via kitchen on CentOS 7.2

### PR DESCRIPTION
**Pull request by GHostQC**
Good day,

This my first contribution to any kind of GitHub project whatsoever, so feel free to give feedback if I'm not doing this the right way.

I was trying to use this cookbook for an elastic stack implementation I'm working for my current employer but our infrastructure is based on CentOS.

I saw in issue #5 that somebody was trying to implement it but hasn't replied to the thread.

So, here is my take to solve issue #5 if you guys are interested in it.

I also implemented ServerSpec test suites since I'm fairly new at "chef-ing" and its currently the only compliance tool I know so far.

Best Regards,

--GHostQC

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 default['elasticsearch-curator']['version'] = nil
 default['elasticsearch-curator']['install_method'] = 'package'
-default['elasticsearch-curator']['repository_url'] = 'http://packages.elastic.co/curator/4/debian'
+default['elasticsearch-curator']['repository_url'] = '' #Default URL are defined in install.rb resource so I can select the one appropriate to the platform (you can still set your own one)
```
Comment (cyberflow) [inline attributes/default.rb]: I think that better define repository url by attributes like:
case node['platform']
when 'centos'
    default['elasticsearch-curator']['repository_url'] = '<repository_url>'
when 'ubuntu'
    default['elasticsearch-curator']['repository_url'] = '<repository_url>'
end

File: resources/install.rb [hunk 2]
```diff
@@ -16,21 +16,42 @@
 action :install do
   if node.platform_family? 'debian'
     apt_repository 'elasticsearch-curator' do
-      uri new_resource.repository_url
+      if new_resource.repository_url == ''
```
Comment (cyberflow) [inline resources/install.rb]: if you will define repository from attributes it's not necessary

**Review by cyberflow — COMMENTED** [review]
Also, can you check and fix general rubocop errors? And thank you very mach for help!

File: resources/install.rb (hunk 2 above)
Comment (GHostQC) [reply resources/install.rb]: Thanks a lot for you feedback !  I will definitely look into it.

**Comment by GHostQC** [comment]
Good day everybody ! 

I have pushed some changes that reflect @cyberflow recommendations.

Travis-CI seems to return errors on %w although every resource on the internet says it should be used with () not [] so I kept it with ()s.  I don't know if there is anything I could do about it.

Also, I had to set the Max Blocklength to 50 instead of 30 as I didn't have any idea on how to make the install action block smaller: [37/30 lines].

Let me know if there is anything else I can do to help !

**Comment by cyberflow** [comment]
@GHostQC Thanks for your help!

**Pull request merged**